extensions/amp-analytics/0.1/events.js

```diff
@@ -306,6 +336,87 @@ export class EventTracker {
   add(unusedContext, unusedEventType, unusedConfig, unusedListener) {}
 }
 
+/**
+ * Tracks custom browser events.
+ */
+export class CustomBrowserEventTracker extends EventTracker {
+  /**
+   * @param {!./analytics-root.AnalyticsRoot} root
+   */
+  constructor(root) {
+    const TIME_WAIT = 500;
+    super(root);
+
+    /** @private {?Observable<!Event>} */
+    this.observables_ = new Observable();
+
+    /** @private {?function(!Event)} */
+    this.boundOnSession_ = this.observables_.fire.bind(this.observables_);
```

Comment:
Just curious, do u really need `bind` here?

Reply:
Yup. Fire by itself does not recieve the event unless bind is added.

Reply:
Re-referencing an instance method makes it lose its `this` binding otherwise.

Reply:
Makes sense. My concern was its a bound function being passed to debounce, which makes it harder to debug & test.